kubernetes/skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFileService.kt

```diff
@@ -61,6 +81,11 @@ class SkaffoldFileService {
      */
     fun findSkaffoldFiles(project: Project): List<VirtualFile> {
         return FileTypeIndex.getFiles(YAMLFileType.YML, GlobalSearchScope.allScope(project))
-            .filter { isSkaffoldFile(it) }
+                .filter { isSkaffoldFile(it) }
     }
+
+    private fun isValidSkaffoldFileType(file: VirtualFile): Boolean =
```

Comment:
This was done as a preliminary check for YAML files, there's nothing particular about Skaffold, how about `isValidYAMLFileType`?

Reply:
oh yeah.. good point. 

I wonder now though if `getSkaffoldVersion` should also first check if its a valid skaffold file too..

Reply:
refactoring now

Reply:
I refactored it to have `getSkaffoldVersion` also check that its a valid skaffold file, while avoiding any duplication of code. 

I'm not sure if I made it worse though..ptal.